test/integration/scheduler/filters/filters_test.go

```diff
@@ -2074,3 +2075,142 @@ func TestUnschedulablePodBecomesSchedulable(t *testing.T) {
 		})
 	}
 }
+
+// TestPodAffinityMatchLabelKeyEnablement tests the Pod is correctly mutated by MatchLabelKeysInPodAffinity feature,
+// even if turing the feature gate enabled or disabled.
+func TestPodAffinityMatchLabelKeyEnablement(t *testing.T) {
```

Comment:
The same is tested manually too, as suggested here: https://github.com/kubernetes/enhancements/pull/4450#discussion_r1480006648